### PR TITLE
feat: add linspace

### PR DESCRIPTION
**Pull request by polvalente**
Brings the implementation from polvalente/nx-signal#3

### CODE REVIEW

File: nx/lib/nx.ex [hunk 1]
```diff
@@ -12084,6 +12084,86 @@ defmodule Nx do
...
+  @doc """
+  Creates a tensor of shape `{n}` with linearly spaced samples between `start` and `stop`
+
+  ## Options
+
+    * `:n` - The number of samples in the tensor.
+    * `:axis_name` - Optional name for the output axis.
+    * `:type` - Optional type for the output. Defaults to `{:f, 32}`
+    * `:endpoint` - Boolean that indicates whether to include `stop`
+                    as the last point in the output. Defaults to `true`
```
Comment (josevalim) [inline nx/lib/nx.ex:12096]:
```suggestion
      as the last point in the output. Defaults to `true`
```

File: nx/lib/nx.ex [hunk 2]
```diff
@@ -12084,6 +12084,86 @@ defmodule Nx do
     apply(impl!(tensor), kind, [out, tensor, opts])
   end
 
+  @doc """
+  Creates a tensor of shape `{n}` with linearly spaced samples between `start` and `stop`
```
Comment (josevalim) [inline nx/lib/nx.ex:12088]:
```suggestion
  Creates a tensor of shape `{n}` with linearly spaced samples between `start` and `stop`.
```

File: nx/lib/nx.ex [hunk 3]
```diff
@@ -12084,6 +12084,86 @@ defmodule Nx do
...
+  ### Error Cases
+
+      iex> Nx.linspace(0, 24, n: nil)
+      ** (ArgumentError) expected n to be a non-negative integer, got: nil
+
+      iex> Nx.linspace(0, 24, n: -1)
+      ** (ArgumentError) expected n to be a non-negative integer, got: -1
+
+      iex> Nx.linspace(0, 24, n: 1.0)
+      ** (ArgumentError) expected n to be a non-negative integer, got: 1.0
```
Comment (josevalim) [inline nx/lib/nx.ex:12140]: One is enough for examples:

```suggestion
      iex> Nx.linspace(0, 24, n: -1)
      ** (ArgumentError) expected n to be a non-negative integer, got: -1
```

Comment (polvalente) [reply nx/lib/nx.ex:12140]: I'd rather keep `1.0` to highlight we only accept plain integers. Will commit separately

**Review by josevalim — COMMENTED** [review]
I dropped some minor comments.

It looks good to me my only question is about `:axis_name`. We don't have a similarly named option anywhere. So I am wondering if we should accept `:names` and force it to be a single-element list or do something else.

Perhaps it is easier to simply remove it for now and someone can call `rename` later if necessary.

File: nx/lib/nx.ex [hunk 4]
```diff
@@ -12084,6 +12084,80 @@ defmodule Nx do
...
 
+  @doc """
+  Creates a tensor of shape `{n}` with linearly spaced samples between `start` and `stop`.
+
+  ## Options
+
+    * `:n` - The number of samples in the tensor.
+    * `:axis_name` - Optional name for the output axis.
+    * `:type` - Optional type for the output. Defaults to `{:f, 32}`
+    * `:endpoint` - Boolean that indicates whether to include `stop`
```
Comment (seanmor5) [inline nx/lib/nx.ex:12095]: Maybe this should be "inclusive" or "include_stop"? Endpoint isn't 100% clear imo

Comment (josevalim) [reply nx/lib/nx.ex:12095]: I like inclusive although I think numpy uses endpoint.

Comment (polvalente) [reply nx/lib/nx.ex:12095]: What about include_endpoint?

**Review by josevalim — APPROVED** [review]

**Pull request merged**